src/sidebar/components/TagListItem.js

```diff
@@ -0,0 +1,68 @@
+import { Icon, Link } from '@hypothesis/frontend-shared';
+import classnames from 'classnames';
+
+/**
+ * @typedef TagListItemProps
+ * @prop {string|null} [href] - If present, tag will be linked to this URL
```

Comment:
I'd suggest to simplify the type to `string` and pass `undefined` to omit it conditionally. This avoids having two ways of indicating that no link is available.